Dprint task fails via VSCode extension on some particular files

Here's the extension Output

[VERBOSE] Checking can format: /Users/stephaneledorze/Documents/projects/rebaze/nx-incubator/libs/filiburg/src/contexts/application.ts
[VERBOSE] Creating instance of dprint-plugin-typescript
[VERBOSE] Created instance of dprint-plugin-typescript in 0ms
[ERROR] Read task failed: Error: Expected success bytes, but found: [99, 111, 109, 101]
[ERROR] Error formatting text. Error: dprint's process exited while the message was in progress.


If I try to format in the repo via `yarn dprint fmt` or if I try `yarn dprint check` they work perfectly

I've tried to reproduce using the initial default config file generated by `dprint init` and it behaves the same

I've failed to shrink the repro as it appears that the problem is not specifically link to a specific part of a file
I can share the file privately however

--- a/src/editor-service/common/EditorProcess.ts
+++ b/src/editor-service/common/EditorProcess.ts
@@ -99,11 +99,28 @@ export class EditorProcess {
   }
 
   async readInt() {
-    const buf = await this.readBuffer(4);
+    const buf = await this.readBufferExact(4);
     return buf.readUInt32BE();
   }
 
-  readBuffer(maxSize: number) {
+  async readBufferExact(size: number) {
+    let i = 0;
+    let finalBuffer: Buffer | undefined;
+    while (i < size) {
+      const remainingSize = size - i;
+      const buffer = await this.readBufferWithMaxSize(remainingSize);
+      if (i == 0 && buffer.length === size) {
+        return buffer;
+      } else if (finalBuffer == null) {
+        finalBuffer = Buffer.alloc(size);
+      }
+      buffer.copy(finalBuffer, i, 0);
+      i += buffer.length;
+    }
+    return finalBuffer ?? Buffer.alloc(0);
+  }
+
+  readBufferWithMaxSize(maxSize: number) {
     this._throwIfNotRunning();
 
     if (maxSize === 0) {

--- a/src/editor-service/implementations/EditorService4.ts
+++ b/src/editor-service/implementations/EditorService4.ts
@@ -70,7 +70,7 @@ export class EditorService4 implements EditorService {
   }
 
   private async assertSuccessBytes() {
-    const buf = await this._process.readBuffer(4);
+    const buf = await this._process.readBufferExact(4);
     if (buf.length !== 4) {
       throw new Error(`Expected success byte array with length 4, but had length ${buf.length}.`);
     }
@@ -121,7 +121,7 @@ async function readString(process: EditorProcess) {
       // send "ready" to CLI
       await writeInt(process, 0);
     }
-    const nextBuffer = await process.readBuffer(stringSize - index);
+    const nextBuffer = await process.readBufferExact(stringSize - index);
     nextBuffer.copy(bytes, index, 0, nextBuffer.length);
     index += nextBuffer.length;
   }

--- a/src/editor-service/implementations/EditorService5.ts
+++ b/src/editor-service/implementations/EditorService5.ts
@@ -34,7 +34,7 @@ export class EditorService5 implements EditorService {
         const messageKind = await this._process.readInt();
         const bodyLength = await this._process.readInt();
 
-        const body = new BodyReader(await this._process.readBuffer(bodyLength));
+        const body = new BodyReader(await this._process.readBufferExact(bodyLength));
         await assertSuccessBytes(this._process);
 
         switch (messageKind) {
@@ -83,7 +83,7 @@ export class EditorService5 implements EditorService {
     }
 
     async function assertSuccessBytes(process: EditorProcess) {
-      const buf = await process.readBuffer(4);
+      const buf = await process.readBufferExact(4);
       if (buf.length !== 4) {
         throw new Error(`Expected success byte array with length 4, but had length ${buf.length}.`);
       }

